src/columns/IPv4.jl

```diff
@@ -0,0 +1,15 @@
+using Sockets: IPv4
+is_ch_type(::Val{:IPv4})  = true
+result_type(::Val{:IPv4})  = Vector{IPv4}
+
+function read_col_data(sock::ClickHouseSock, num_rows::VarUInt, ::Val{:IPv4})
+    tmp = Vector{IPv4}(undef, UInt64(num_rows))
+    Base.read!(sock.io, tmp)
```

Comment:
We can add them, but IMHO it is redundant. These are basic types (as well as UUIDs) that are defined by the byte representation described in the standards. All kinds of stock output capabilities, etc. this is syntactic sugar. Then we need the same checks on Int32, because developers can add a field there as well

Reply:
As an option, I can suggest sending the UUID.value and IPv4/6.host   instead of the structures themselves. These parameters must exactly follow the standards for the corresponding entities

Reply:
Well yeah, it's more of a general problem of the library and the assert won't really fix it anyway because layout can also vary without any padding occurring / within the same size.

They might have a standard representation for how they are supposed to be serialized over network but that doesn't mean that necessarily matches their in-memory layout. `Base.write(stream, ::Array{T})` will just take whatever buffer you hand it and write it out to the socket precisely as it finds it in memory as long as `isbitstype(T) == true`. The exact memory layout, however, depends on a lot of factors -- the most important one being CPU architecture. 

You are in fact actually right that this also applies to `Int32` -- our whole current code will, for example, not work with the PowerPC version of Julia because PPC has big-endian memory layout and we will send out the ints as BE when CH expects LE -- however this is broken all over the code-base and hardly anyone uses BE architecture CPUs. 

Quoting from the documentation of `Base.write`:
> The endianness of the written value depends on the endianness of the host system. Convert to/from a fixed endianness when writing/reading (e.g. using htol and ltoh) to get results that are consistent across platforms.

I guess fixing it in this one place now won't help a lot -- it's probably better to leave things as-is for now and fix this all over the codebase in a different PR. Hardly anyone uses BE architectures anyway.

Reply:
Among other things, I don't remember seeing any reencoding of byte order in the Сlickhouse code before sending response. If I didn't miss anything, then this means that it is most likely determined not by the network, but by the byte order of the system on which it is running. That is, the native client, as well as replication and clustering will not work if they are running on systems with different byte order. I think the same restriction for our client looks reasonable at least at the moment

Reply:
Yeah, agreed!